lib/ssl/src/ssl_gen_statem.erl

```diff
@@ -141,44 +140,46 @@ start_link(Role, Host, Port, Socket, {SslOpts, _, _} = Options, User, CbInfo) ->
 -spec init(list()) -> no_return().
 %% Description: Initialization
 %%--------------------------------------------------------------------
-init([Role, _Sender, Host, Port, _Socket, {TLSOpts, _, _},  _User, _CbInfo] = InitArgs) ->
+init([Role, Sender |[Host, Port, _Socket, {TLSOpts, _, _}, _User, _CbInfo] = InitArgs]) ->
     process_flag(trap_exit, true),
 
     case maps:get(erl_dist, TLSOpts, false) of
         true ->
             process_flag(priority, max);
         _ ->
             ok
-    end,    
-    
+    end,
+
     init_label(Role, Host, Port, TLSOpts),
+    Tab = ets:new(tls_socket, []),
```

Comment:
tls_socket  is the name of the module implementing transport functions for TLS and dtls_socket does the same for DTLS. These functions are used so that the transport functionality should be pluggable. sslsocket (is legacy record name) for handle that is matchable but should be handled as opaque by the user.